Revise Install Guide to provide better, more up to date guidance

CAM-Gerlach opened the pull request:
<!--- Before submitting your pull request, --->
<!--- please complete as much as possible of the following checklist: --->

# Pull Request

## Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder-docs/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 4.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed


## Description of Changes

<!--- Describe what you've changed and why. --->

In response to the detailed user feedback on #334 , this PR:

* Expands and updates the guidance in the installation guide surrounding the standalone installers and Anaconda/Miniconda
* Adds more details about the standalone installers, their limitations and how to work around them,
* Adjusts the recommendations accordingly
* Adds a description of installing via Miniconda rather than Anaconda, and guides users toward it
* Makes other related smaller improvements

Additionally, I've revised and updated the other sections in the install guide to ensure they are complete, up to date, and easy to follow.

[Live preview of updated install guide](https://deploy-preview-335--spyder-docs-preview.netlify.app/current/installation.html)


### Issue(s) Resolved

<!--- Pull requests should typically resolve at least one—preferably only one— --->
<!--- outstanding issue; create a new one if no relevant issue exists. --->
<!--- List the issue(s) below, in the form "Fixes #1234" . One per line. --->
<!--- However, smaller fixes, maintenance or trivial changes don't need one. --->

Fixes #334


<!--- Thanks for your help making Spyder --->
<!--- and its documentation better for everyone! --->

jitseniesen reviewed (commented):
Looks good to me; just one specific suggestion and one more general question: Should we recommend [Miniforge / Mambaforge](https://github.com/conda-forge/miniforge) instead of Miniconda?

ccordoba12 commented on doc/installation.rst:
```diff
@@ -85,57 +94,73 @@ On macOS, the first time you open Spyder, you may see a message that it cannot b
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To install Spyder's optional dependencies as well for full functionality, use the following command instead:
+Once you download and install Miniconda, or to get a more reliable and up-to-date Spyder install with Anaconda, install Spyder into its own fresh environment.
```
```suggestion
Once you download and install Miniconda, or to get a more reliable and up-to-date Spyder version with Anaconda, we recommend installing Spyder into its own fresh environment.
```

* Avoid repeating `install` three times in this sentence.
* Mention explicitly that we recommend this.

CAM-Gerlach replied:
> Avoid repeating install three times in this sentence.

Yeah, I actually already rewrote this sentence to avoid that.

> Mention explicitly that we recommend this.

Reworded a bit, but I implemented the intent here

ccordoba12 commented on doc/installation.rst:
```diff
@@ -85,57 +94,73 @@ On macOS, the first time you open Spyder, you may see a message that it cannot b
+   conda create -c conda-forge -n spyder-env spyder
 
-You can then run Spyder by the same methods as above, except make sure to select the start menu shortcut with ``(spyder-env)`` in the name, select the ``spyder-env`` environment on the left before launching it with Navigator, or type ``conda activate spyder-env`` before launching it on the command line.
+This installs Spyder into a new environment called ``spyder-env``, using the more up-to-date, community Conda-Forge channel.
```
```suggestion
This will install Spyder into a new environment called ``spyder-env``, using the more up-to-date, community Conda-Forge channel.
```

I'm not sure if this correct. If not, please dismiss it.

CAM-Gerlach replied:
Either is perfectly valid, but "installs" is a bit more concise and direct, so I didn't end up applying this.

ccordoba12 reviewed (commented):
Thanks @CAM-Gerlach for your work on this!

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -85,57 +94,73 @@ On macOS, the first time you open Spyder, you may see a message that it cannot b
 
-Spyder is included by default in the `Anaconda Python distribution`_, which comes with everything you need to get started in an all-in-one package, and is our recommended installation method on Linux (and supported on the other platforms too).
+Spyder is included by default in the `Anaconda`_ Python distribution, which comes with everything you need to get started in an all-in-one package, and can easily be installed in the much lighter-weight `Miniconda`_, which includes just Python and the Conda package and environment manager by default.
+This is our recommended installation method on Linux and for users needing third-party Spyder plugins, and supported on Windows and macOS too.
```
I've substantially revised this in the current rewritten version, to avoid this clunkiness and clarify why the installer is recommended.

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -11,10 +11,10 @@ Thanks!
 
-   Spyder now offers :ref:`standalone installers<standalone_installers_ref>` for Windows and macOS, making it easier to get up and running with the application without having to download Anaconda or manually install it in your existing environment.
-   While we still support Anaconda, we recommend this install method on those platforms to avoid most problems with package conflicts and other issues.
+   For most users on Windows and macOS, we recommend using our :ref:`standalone_installers_ref`, and for those on Linux or who need third-party Spyder plugins, which the standalone installers currently do not support, we advise using :ref:`installation-conda`.
```
> Say why we recommend our installers.

I actually included such in several previous drafts, but I didn't want to duplicate the information in the standalone installers section, so I ended up eliding it in the end. However, I revised it to mention this again (with more appropriate diction).

> Mention other packages because that's important for users.

The specific limitation here with the standalone installers and third party packages is the ability to view the latter's data types in the Variable Explorer (assuming a solution to that hasn't been implemented yet), which I added a mention of.

Otherwise, for non-included packages, users just have to use a separate Python environment to run their code (provided by an existing or new Anaconda/Miniconda/etc installation, in this case), which is what we've always told them to do anyway for years and years now to avoid breaking Spyder, and is the whole point of the standalone installers. This is described in more detail, with a link to our FAQ and How-To video on the subject, at the top of the standalone installers section, and I also include more guidance on setting this up in the Conda section.

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -11,10 +11,10 @@ Thanks!
-   Spyder now offers :ref:`standalone installers<standalone_installers_ref>` for Windows and macOS, making it easier to get up and running with the application without having to download Anaconda or manually install it in your existing environment.
-   While we still support Anaconda, we recommend this install method on those platforms to avoid most problems with package conflicts and other issues.
+   For most users on Windows and macOS, we recommend using our :ref:`standalone_installers_ref`, and for those on Linux or who need third-party Spyder plugins, which the standalone installers currently do not support, we advise using :ref:`installation-conda`.
+   Linux, plugin and package/environment management support in the standalone installers are currently under active development for future Spyder versions.
```
I wanted the order to match the previous sentence, where the plugin support (which is the bigger deal there) is mentioned second. Its a bigger feature here, but its more of a nice to have than resolving a hard limitation in the installers like plugin support.

Also, I'm not sure its 100% set that it will come first, we could just add support for selecting custom plugins at install time, add a CLI or simplified minimal GUI for installing plugins before adding full package and env management for arbitrary packages and environments, which is a much larger problem space.

ccordoba12 replied:
> I wanted the order to match the previous sentence

Ok, I understand now the idea of this sentence. So, I'm ok with leaving it as it is.

> Also, I'm not sure its 100% set that it will come first, we could just add support for selecting custom plugins at install time, add a CLI or simplified minimal GUI for installing plugins before adding full package and env management for arbitrary packages and environments, which is a much larger problem space.

Package management will come first thanks to what Daniel and Juan Sebastian are doing.

CAM-Gerlach replied:
Okay, I see. In any case, if we change it here, we would need to change it in the previous sentence as well, and I'm really not sure it makes sense to put better support for custom data types in the variable explorer ahead of plugin support, especially when its not really that directly connected with the package and environment management anyway (as it refers to Spyder's own internal environment).

ccordoba12 replied:
As I said above, I'm fine with leaving this sentence as it is.

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -40,15 +40,24 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
-Our standalone installers for Windows and macOS are available from Spyder 4.2 onwards.
-We recommend using this installation method on those platforms, but we offer several other options for Linux, advanced users and specific needs, so keep reading if that's the case for you.
+The standalone installers are our recommended method for most users on Windows and macOS, with experimental Linux support under active development.
+They work much like installing any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code, avoiding problems with incompatible packages and botched installations that users often face when doing so within their own Python installation.
```
Good point; should have thought of that. Implemented.

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -85,57 +94,73 @@ On macOS, the first time you open Spyder, you may see a message that it cannot b
-New Conda environment
-~~~~~~~~~~~~~~~~~~~~~
+   The bundled Spyder version can often be quite out of date, missing new features and bug fixes from the latest version, and if you install, change or remove other packages, there is a chance of dependency conflicts or a broken Spyder installation.
+   We recommend :ref:`installing Spyder into a new Conda environment <installation-miniconda>` to avoid all these issues.
```
Funny, I'm pretty sure I had that in an earlier draft, but removed it for some reason. Added it back.

CAM-Gerlach commented:
In the process of responding to @jitseniesen 's comments, I ended up noticing a bunch of things that were out of date, suboptimal or could be clarified/expanded in the other sections too, so I went ahead and revised them here as well. Due that, I've manually applied the remaining relevant comments from @ccordoba12 with various fixes, updates and adaptations, since most of them are now out of date.

> Should we recommend [Miniforge / Mambaforge](https://github.com/conda-forge/miniforge) instead of Miniconda?

I thought about that before, actually, but wasn't sure about it at the time.

On one hand it would avoid needing to tell users to explicitly specify the Conda-Forge channel when installing Spyder, and avoids problems when they update it and forget to specify it (but that can be fixed by adding a `conda config --env --add channels conda-forge` command). Mamba would also speed things up, (but `libmamba` is now available as a solver for `conda` thanks to Felipe's work, and users may not be as familiar with it).

On the other hand, we certainly still want to mention Miniconda, since as far as I'm aware it is still a lot more popular than Miniforge/Mambaforge, has a much friendlier website and documentation, and we want to support people with their existing environment.

With that in mind, I rewrote and updated the Anaconda/Miniconda section to refer to Conda installs more generally, mentioning the various distributions including those. This also inspired me to update the other sections as well, as mentioned above.

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
+By default, Spyder checks for updates automatically on startup.
+You can also do manually with :menuselection:`Help --> Check for updates` on Windows and Linux, or :menuselection:`Spyder --> Check for updates` on macOS.
+
+The :ref:`install-standalone` for Spyder 5.4.0+ also include update functionality built right into Spyder, which after checking for updates as above will display a prompt to automatically download and install the current version.
```
@mrclary Is this true on macOS as well? Or did we not add that support in favor of just waiting for the `conda`- based installers? I can't remember the final outcome, sorry...

mrclary replied:
The menu selection on macOS is `Help --> Check for updates`, same as Windows and Linux (should it be different?).

And, yes, the update mechanism works on macOS. The only substantive difference is that macOS users will have to manually drag-and-drop the application.

CAM-Gerlach replied:
Oh, I thought the canonical location for update checks on macOS was the *<Application>* menu? Partially for that reason, I always thought it was really odd that it Windows applications typically put it under the `Help` menu of all places when I switched to Windows back when I was younger.

Anyway, I've applied a suggestion to update this.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -11,12 +13,14 @@ Thanks!
-   Spyder now offers :ref:`standalone installers<standalone_installers_ref>` for Windows and macOS, making it easier to get up and running with the application without having to download Anaconda or manually install it in your existing environment.
-   While we still support Anaconda, we recommend this install method on those platforms to avoid most problems with package conflicts and other issues.
+   For most users on Windows and macOS, we recommend our :ref:`install-standalone` as the most straightforward and robust option to obtain Spyder.
+   For users on Linux, needing third-party Spyder plugins or Variable Explorer compatibility with custom-installed packages, capabilities which that the standalone installers currently do not provide, we advise using a :ref:`install-conda`.
```
```suggestion
   For users on Linux, needing third-party Spyder plugins or Variable Explorer compatibility with custom-installed packages, capabilities which the standalone installers currently do not provide, we advise using a :ref:`install-conda`.
```

ccordoba12 commented on doc/installation.rst:
```diff
@@ -34,25 +38,35 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
-Our standalone installers for Windows and macOS are available from Spyder 4.2 onwards.
-We recommend using this installation method on those platforms, but we offer several other options for Linux, advanced users and specific needs, so keep reading if that's the case for you.
+The standalone installers are our recommended method for most users on Windows and macOS, with experimental Linux support under active development.
+They work much like installing any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code, avoiding problems with incompatible packages and broken installations that users often face when doing so within their own Python installation.
```
```suggestion
They work much like installing any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code, avoiding problems with incompatible packages and broken installations that users often face when doing so within Anaconda, for instance.
```

* Avoids repeating `installation` twice.
* I think it provides a simpler reference (i.e. Anaconda) for users to understand what we mean here.

CAM-Gerlach replied:
Okay, though its a little weird to say "for instance" here. I've reworded it accordingly, and also split it into two shorter sentences and avoided another repeat of "installing".

```suggestion
They work much like for any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code.
This avoids the problems with incompatible packages and broken installations users often face when mixing Spyder with the (Conda, etc) environments they use to run their code.
```

ccordoba12 commented on doc/installation.rst:
```diff
@@ -34,25 +38,35 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
+The standalone installers are our recommended method for most users on Windows and macOS, with experimental Linux support under active development.
+They work much like installing any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code, avoiding problems with incompatible packages and broken installations that users often face when doing so within their own Python installation.
+
+The installers include a built-in Python environment with the most common scientific libraries, which can be used out of the box for basic data analysis tasks.
```
```suggestion
The installers include a built-in Python environment with the most common scientific libraries (e.g. Matplotlib Pandas and Numpy), which can be used out of the box for basic data analysis tasks.
```

Just so users get a feeling of what they're going to find in the installers.

CAM-Gerlach replied:
```suggestion
The installers include a built-in Python environment with the most common scientific libraries (e.g. NumPy, Pandas, Matpotlib, etc), which can be used out of the box for basic data analysis tasks.
```

Fix punctuation, capitalization and ordering

ccordoba12 commented on doc/installation.rst:
```diff
@@ -34,25 +38,35 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
+They work much like installing any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code, avoiding problems with incompatible packages and broken installations that users often face when doing so within their own Python installation.
+
+The installers include a built-in Python environment with the most common scientific libraries, which can be used out of the box for basic data analysis tasks.
+However, to install and manage your own packages and environments, you'll currently need to use an external Python installation (such as `Anaconda`_, `Miniconda`_, `Miniforge/Mambaforge`_, `WinPython`_ or `Python.org <Python_>`__) with Spyder.
```
```suggestion
However, to install and manage your own packages and environments, you'll currently need to use an external Python installation (such as `Anaconda`_, `Miniconda`_, `Miniforge/Mambaforge`_, `WinPython`_ or `Python.org <Python_>`__), which needs to be connected to our installers.
```

I think `with Spyder` would lead people to install Spyder on that installation, which we know is not necessary.

CAM-Gerlach replied:
```suggestion
However, to manage your own packages and environments, you'll currently need to connect an external Python distribution (such as `Anaconda`_, `Miniconda`_, `Miniforge/Mambaforge`_, `WinPython`_ or `Python.org <Python_>`__) to your standalone copy of Spyder.
```

* Avoid triple-repeat of installation
* Be more concise/less redundant

ccordoba12 commented on doc/installation.rst:
```diff
@@ -34,25 +38,35 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
+
+.. note::
+
+   The standalone installers currently do not yet support installing third-party Spyder plugins not already bundled with them, though this feature is currently under development.
```
```suggestion
   The standalone installers currently do not yet support installing third-party Spyder plugins not already bundled with them, though this feature is under development.
```

`currently` is said twice here.

CAM-Gerlach replied:
```suggestion
   The standalone installers do not yet support installing third-party Spyder plugins not already bundled with them, though this feature is currently under development.
```

Yup, though its the other "currently" that should go, as "currently" and "do not yet" say the same thing.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -76,148 +90,191 @@ On macOS, open the disk image and drag Spyder to your :guilabel:`Applications` f
+
+   If using Mamba/Mambaforge, substitute ``mamba`` for ``conda`` in the following commands.
+
+With Miniconda/Miniconda/Miniforge, or to get a more reliable and up-to-date Spyder version with Anaconda, Spyder should be installed into its own dedicated Conda environment.
```
```suggestion
With Miniconda/Mambaforge/Miniforge, or to get a more reliable and up-to-date Spyder version with Anaconda, Spyder should be installed into its own dedicated Conda environment.
```

CAM-Gerlach replied:
```suggestion
With Miniconda/Miniforge/Mambaforge, or to get a more reliable and up-to-date Spyder version with Anaconda, Spyder should be installed into its own dedicated Conda environment.
```

Good catch, but they should be in the correct order

ccordoba12 commented on doc/installation.rst:
```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
    :alt: Running Spyder installation with pip
 
-You may need to install a Qt binding (PyQt5) separately with ``pip`` if running under Python 2.
+To launch Spyder after installing, ensure your environment is activated and run the ``spyder`` or ``spyder3`` command.
```
```suggestion
To launch Spyder after installing it, ensure your environment is activated and run the ``spyder`` or ``spyder3`` command.
```

Seems missing.

CAM-Gerlach replied:
You _can_ get away with eliding it, but better to have it I guess

ccordoba12 commented on doc/installation.rst:
```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
-.. code-block:: bash
+.. code-block:: shell
 
-   conda update anaconda
```
Most of the time users get a package resolution conflicts without running `conda update anaconda` first. So, we should keep this command.

Furthermore, these commands could also fail sometimes with the same kind of error. In that case, they've told us they need to run

    conda remove spyder
    conda install spyder

CAM-Gerlach replied:
Well, that's only true of Anaconda, whereas `conda update anaconda` will generally fail with an error on the other three, as the `anaconda` metapackage will generally not be installed. And are you sure that workaround works reliably? Wouldn't it be better to just tell users to re-solve/re-create their environment (and probably faster too, because it avoids two resolution steps and a clean solve is much faster than an incremental one)?

ccordoba12 replied:
> Well, that's only true of Anaconda

That's right. Then we'd need to differentiate at this point between updating Spyder in Anaconda and Miniconda, Miniforge, etc.

> And are you sure that workaround works reliably?

Yes, and it's also important to mention the second workaround I posted above.

> Wouldn't it be better to just tell users to re-solve/re-create their environment (and probably faster too, because it avoids two resolution steps and a clean solve is much faster than an incremental one)?

That's not a possibility for `base` and the problem with installing Spyder in other envs is that you have to connect it to `base` and keep track that the right version of Spyder-kernels is installed on it.

CAM-Gerlach replied:
Thanks:

* I thoroughly revised the Updating section to incorporate these improvements and others.
* It then was long enough that I re-organized it and split it into subsections per method to be easier to follow.
* It it turn occurred to me that it would make more sense just putting it under the respective top-level method headings, like the "Running Spyder" sections are, so everything for that method is in one place.
* That then motivated me to re-organize the Conda section into separate subsections for Conda envs and base Anaconda, and to put the former before the latter to further emphasize it.
* Finally, I also moved the "Alternative Methods" section below the pip section, as unlike pip they are third party and we don't control them at all, so better to make more clear that fact.

ccordoba12 commented on doc/installation.rst:
```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
-This command will also update all Spyder dependencies, so we recommend you use an isolated virtual environment to avoid any potential unintended effects on other installed packages.
+.. note::
 
+   This command will also update all Spyder dependencies, so we recommend you use an isolated virtual environment to avoid any potential unintended effects on other installed packages.
```
```suggestion
   This command will also update all Spyder dependencies, so we recommend you use an isolated conda environment to avoid any potential unintended effects on other installed packages.
```

ccordoba12 reviewed (commented):
Thanks for the new batch of updates @CAM-Gerlach! This is looking pretty good!

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -257,54 +314,76 @@ You'll first want to create and activate a virtual environment in which to insta
 ===============
 
-To update Spyder installed via our :ref:`standalone packages<standalone_installers_ref>` on Windows and macOS, you'll currently need to manually download and install the latest release (if on Windows, make sure to remove the old version first from Control Panel/System Settings).
+By default, Spyder checks for updates automatically on startup.
+You can also do manually with :menuselection:`Help --> Check for updates` on Windows and Linux, or :menuselection:`Spyder --> Check for updates` on macOS.
```
```suggestion
By default, Spyder checks for updates automatically on startup, and you can also check manually with :menuselection:`Help --> Check for updates`.
```

Updated this based on @mrclary 's input below.

mrclary commented on doc/installation.rst:
```diff
@@ -76,148 +91,191 @@ On macOS, open the disk image and drag Spyder to your :guilabel:`Applications` f
+To run Spyder when installed standalone, you can simply use your operating system's typical method of launching applications, such as opening it from the :guilabel:`Start` menu on Windows (or the Taskbar, if you've pinned it there), or from Launchpad, Spotlight or the :guilabel:`Applications` folder on macOS (or the Dock, if you've added it there).
 
-On macOS, the first time you open Spyder, you may see a message that it cannot be opened because the developer cannot be verified.
+On macOS, the first time you open Spyder, you may see a message that it cannot be opened because "the developer cannot be verified".
 If so, right-click on the application, select :guilabel:`Open`, then click :guilabel:`Open` in the resulting dialog, and the warning will no longer be shown.
 You can also click :guilabel:`Open Anyway` under :menuselection:`Security & Privacy --> General` in System Preferences.
```
```suggestion
```
Our macOS installers are codesigned and notarized, so the user should never encounter this scenario.

CAM-Gerlach replied:
Thanks; I wasn't sure if this was still a possibility in some scenarios.

mrclary reviewed (commented):
Should we mention the experimental conda-based installers at all, since they will appear next to the other installers as 5.4.0 release artifacts?

CAM-Gerlach commented:
@ccordoba12 any chance you could take another look at this? Also, @jitseniesen , would you to review this as well?

ccordoba12 commented on doc/installation.rst:
```diff
@@ -76,176 +91,183 @@ On macOS, open the disk image and drag Spyder to your :guilabel:`Applications` f
+-------------------
 
-.. _MacPorts project: https://www.macports.org/
+With any Conda-based distribution and Spyder installed in its own environment (recommended), update Conda itself, active the environment then update Spyder.
```
```suggestion
With any Conda-based distribution and Spyder installed in its own environment (recommended), update Conda itself, active the environment and finally update Spyder.
```

I think this is easier to read.

CAM-Gerlach replied:
Sure, just add a comma (Oxford or not) to be clearer about the sequence/semantics:

```suggestion
With any Conda-based distribution and Spyder installed in its own environment (recommended), update Conda itself, active the environment, and finally update Spyder.
```

ccordoba12 commented on doc/installation.rst:
```diff
@@ -76,176 +91,183 @@ On macOS, open the disk image and drag Spyder to your :guilabel:`Applications` f
 
-Spyder is included in the `WinPython`_ scientific Python distribution (although Anaconda's ``conda`` package and environment manager is not).
-You can use Spyder immediately after installing, just like with Anaconda.
+You can then run Spyder by the same methods :ref:`as with Anaconda <install-anaconda-running>`, except make sure to launch the start menu shortcut with ``(spyder-env)`` in the name, select the ``spyder-env`` environment on the left before launching it with Navigator, or type ``conda activate spyder-env`` before launching it on the command line.
```
```suggestion
You can then run Spyder by the same methods :ref:`as with Anaconda <install-anaconda-running>`, except that you need to make sure to launch the Start menu shortcut with ``(spyder-env)`` in the name, select the ``spyder-env`` environment on the left before launching it with Navigator, or type ``conda activate spyder-env`` before launching it on the command line.
```

* I think this would make the sentence easier to read for non-native English speakers, although I'm not sure if it is correct.
* Start menu is capitalized in other places (and I think that's correct).

CAM-Gerlach replied:
Sure; its a little belabored, but it is perfectly correct and clearer to me as well :+1:

ccordoba12 commented on doc/installation.rst:
```diff
@@ -76,176 +91,183 @@ On macOS, open the disk image and drag Spyder to your :guilabel:`Applications` f
+.. caution::
 
-.. code-block:: bash
+   The bundled Spyder version can often be quite out of date, missing new features and bug fixes from the latest version, and if you install, change or remove other packages, there is a chance of dependency conflicts or a broken Spyder installation.
```
```suggestion
   The bundled Spyder version can often be quite out of date, missing new features and bug fixes from the latest version, and if you install, change or remove other packages, there is an important chance of dependency conflicts or a broken Spyder installation.
```

To discourage people even more from using Spyder from the base env,

CAM-Gerlach replied:
"an important chance" doesn't make any semantic sense here in English. However, we can use a more appropriate word for emphasis:

```suggestion
   The bundled Spyder version can often be quite out of date, missing new features and bug fixes from the latest version, and if you install, change or remove other packages, there is a significant chance of dependency conflicts or a broken Spyder installation.
```

ccordoba12 reviewed (commented):
Thanks for the ping @CAM-Gerlach! I left some small suggestions, otherwise looks ready to be merged.

mrclary commented on doc/installation.rst:
```diff
@@ -34,25 +38,36 @@ Just visit the `Spyder page on Binder`_ to get started using Spyder now.
-Our standalone installers for Windows and macOS are available from Spyder 4.2 onwards.
-We recommend using this installation method on those platforms, but we offer several other options for Linux, advanced users and specific needs, so keep reading if that's the case for you.
+The standalone installers are our recommended method for most users on Windows and macOS, with experimental Linux support under active development.
+They work much like for any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code.
```
```suggestion
They work similar to any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code.
```

CAM-Gerlach replied:
Technically, the revised version implies that the installers work similar to other IDEs themselves rather than those IDEs' _installers_. If we're going to go this route, may as well make it even simpler:

```suggestion
They work like any other IDE, where Spyder can be installed and updated independently of the Python environments you use to run your code.
```

mrclary reviewed (commented):
Just a few comments; otherwise LGTM

CAM-Gerlach commented on doc/installation.rst:
```diff
@@ -11,12 +13,14 @@ Thanks!
-   Spyder now offers :ref:`standalone installers<standalone_installers_ref>` for Windows and macOS, making it easier to get up and running with the application without having to download Anaconda or manually install it in your existing environment.
-   While we still support Anaconda, we recommend this install method on those platforms to avoid most problems with package conflicts and other issues.
+   For most users on Windows and macOS, we recommend our :ref:`install-standalone` as the most straightforward and robust option to obtain Spyder.
+   For users on Linux, needing third-party Spyder plugins or Variable Explorer compatibility with custom-installed packages, capabilities which the standalone installers currently do not provide, we advise using a :ref:`install-conda`.
```
```suggestion
   For users needing Linux support, third-party Spyder plugins or Variable Explorer compatibility with custom-installed packages—all capabilities which the standalone installers currently do not yet provide—we advise using a :ref:`install-conda`.
```

Instead of repeating users again as well as a doubled-up "or", I rewrote the sentence to make it clearer and simpler and avoid the need for that.

ccordoba12 reviewed (approved):
Looks good to me now. Thanks @CAM-Gerlach for the improvements, great work!

CAM-Gerlach commented:
Thanks all for your great feedback! I did a bit of rebasing to clean up the commits for merging and give you both credit for your suggestions; otherwise looks like this is good to go (since we haven't heard from Jitse in a bit, and his feedback was implemented), and so I can switch to squash merge only to make sure that #331 isn't accidentally non-squashed with the large images/GIFs.
